plugins/Login/PasswordResetter.php

```diff
@@ -383,7 +395,11 @@ protected function getUserInformation($loginOrMail)
      */
     protected function checkPasswordHash($passwordHash)
     {
-        UsersManager::checkPasswordHash($passwordHash, Piwik::translate('Login_ExceptionPasswordMD5HashExpected'));
+        $hashInfo = $this->passwordHelper->info($passwordHash);
+
+        if (!isset($hashInfo['algo']) || 0 >= $hashInfo['algo']) {
+            throw new Exception(Piwik::translate('Login_ExceptionPasswordMD5HashExpected'));
```

Comment:
Just saw the screenshot was failing: http://builds-artifacts.piwik.org/piwik/piwik/3.x-dev/21123/Login_password_reset_complete.png

After modifying the \UsersManager\Model to accept only fully hashed passwords I noticed the resetter still stored the new password not fully hashed. However after updating `savePasswordResetInfo()` the old sanity check still expected a 32 char hash.

I changed the sanity check to require a string compatible with `password_hash()`. This should be safe even when the default algorithm changes (sometime far in the future...).